### ISSUE TITLE
Site labels / translation

### ISSUE THREAD

**Issue by tobi-or-not-tobi** [SAP/cloud-commerce-spartacus-storefront#9]
The storefronts is driven by different content streams. Most of the content is driven by either the CMS or by the product content. There's however a portion of content that is provided by so-called site labels.

Site labels are in the current storefront solution (spring mvc based) based on java resource bundles, which gives some issues:
- they cannot be controlled by business users and require a full development cycle
- they aren't exposed through a webservice
- they do not have any structure (it's a flat list, although they're separated over multiple files)

There's a lot of room for improvement. We like to setup a solution that provides:
- runtime language switching
- lazy load languages
- avoid hardcoded text

Angular doesn't provide these features yet, but have hired the author of translate to build this is as a standard feature in Angular. This takes longer then expected and it's recommended to start embracing [ng-translate](https://github.com/ngx-translate/core) so that we can relatively smoothly transition to the standard solution (probably beyond ng7 and in parallel with ngyvi).

**Comment by Platonn** [comment]
### Prerequisites to investigate before we decide to use `ngx-translate`:
- can we use `ngx-translate` library both in shell app and in the library? Doesn't it cause any problems?
- can we use `ngx-translate` library together with SSR? 

### Acceptance criteria - when we are done:
- no wordings are hardcoded in our code - all taken from JSON with translations
- translation mechanism used everywhere:
   - in components' templates
   - in the code
- we provide a few JSON files with translations in English (for Spartacus)
- Customer can provide JSONs for other languages (for Spartacus)
- Customer can overwrite our keys or extend (introduce new keys) with his own JSONs
- Customer can reuse our keys in his custom components
- Customer's lazy loaded components can use it's own isolated translations
- current language used for translations is taken from site context
- Customer can provide JSONs with translations for his custom components, by using the same library as us - `ngx-translate`

### Other research:
- how to support plural forms of translations using `ngx-translate`
- how to structure the keys of translations and split into separate files - we should avoid breaking changes of the strucutre in the future
- what are best practices for migrating from `ngx-translate` to the native `i18n` baked in the Ivy

### Assumptions:
- JSON files are just static assets served by the the shell app

### Possible future outlook:
- fetching translations from some occ endpoint and manage in CMS

### Out of scope (beacause Ivy should be relased in the end of 2019):
- migrate from `ngx-translate` to the native `i18n` baked in the Ivy

### Questions:
- should we provide JSON with translations for other languages before Launch 1.0?

**Comment by Xymmer** [comment]
i closed this ticket as a dupe of all the localization efforts, but will reopen if needed
